### ISSUE TITLE
Broken snowflake_database resource in 0.60.0 provider version

### ISSUE THREAD

**Issue by marcincuber** [Snowflake-Labs/terraform-provider-snowflake#1674]
**Provider Version**

0.60.0

**Terraform Version**

1.4.1

**Describe the bug**

Snowflake databases seem to disappeared from state file and terraform plan wants to re-create already deployed databases.

**Expected behavior**

Database should exists and plan should report no changes.

**Code samples and commands**

```
resource "snowflake_database" "eurowind" {
  name = "EUROWIND"
}
```

**Additional context**

Another problem is that with this provider version, even import command doesn't work.
```
terraform import snowflake_database.eurowind EUROWIND                                                                                                                          
╷
│ Error: Cannot import non-existent remote object
│
│ While attempting to import an existing object to "snowflake_database.eurowind", the provider detected that no object exists with the given id. Only pre-existing objects can be imported; check that the id is
│ correct and that it is associated with the provider's configured region or endpoint, or use "terraform apply" to create a new remote object for this resource.
```

This release is super broken. Fix was to downgrade to provider and re-import database resources.

**Comment by mlorek** [comment]
also getting the following err on resource monitor options change
11:37:20  
11:37:20  Stack trace from the terraform-provider-snowflake_v0.60.0 plugin:
11:37:20  
11:37:20  panic: interface conversion: interface {} is *schema.Set, not int

**Comment by adamhealy** [comment]
+1 on this issue. Same issue with databases, same fix to roll back an re-import

**Comment by kietay** [comment]
If the provider fails to query the dbs it just thinks the resource has been deleted. It should be checking for a nil, not an error https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/pkg/resources/database.go#L231-L235

**Comment by sfc-gh-swinkler** [comment]
@marcincuber what is the ID of your database resource? I would like to help resolve this issue, but I cannot replicate the issue in the 0.61 version. 

configuration:
```
resource "snowflake_database" "eurowind" {
  name = "EUROWIND"
}

```

apply is succesful
```
snowflake_database.eurowind: Creating...
snowflake_database.eurowind: Creation complete after 1s [id=EUROWIND]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

no changes during subsequent plan

```
snowflake_database.eurowind: Refreshing state... [id=EUROWIND]

No changes. Your infrastructure matches the configuration.
```

removing the resource from TF state and reimporting:

```
$ terraform state rm snowflake_database.eurowind
Removed snowflake_database.eurowind
Successfully removed 1 resource instance(s).
$ terraform import snowflake_database.eurowind EUROWIND
snowflake_database.eurowind: Importing from ID "EUROWIND"...
snowflake_database.eurowind: Import prepared!
  Prepared snowflake_database for import
snowflake_database.eurowind: Refreshing state... [id=EUROWIND]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.
```

final plan

```
$ terraform plan
╷
│ Warning: Provider development overrides are in effect
│
│ The following provider development overrides are set in the CLI
│ configuration:
│  - snowflake-labs/snowflake in /Users/swinkler/go/bin
│
│ The behavior may therefore not match any released version of the provider and
│ applying changes may cause the state to become incompatible with published
│ releases.
╵
snowflake_database.eurowind: Refreshing state... [id=EUROWIND]

No changes. Your infrastructure matches the configuration.
```

**Comment by sfc-gh-swinkler** [comment]
> Something seems off.. What is the 0.34.0 tag all about? <img alt="Screenshot 2023-04-03 at 14 01 03" width="992" src="https://user-images.githubusercontent.com/50456/229590207-900b0c9b-453f-40b6-8938-a57de930fb4d.png">

Just a quirk in the release-please automation. I am not sure why it does that

**Comment by marcincuber** [comment]
@sfc-gh-swinkler I will give it a try with the latest 0.61 provider. However, have you tested it with 0.60.0? It is a release which is clearly broken, I would even suggest removing it. In my case, it fundamentally broke three state files with many databases.

**Comment by marcincuber** [comment]
@sfc-gh-swinkler just to confirm. Databases are not detected again with 0.61. This is confirmed by running a plan:
```
Terraform will perform the following actions:

  # snowflake_database.eurowind will be created
  + resource "snowflake_database" "eurowind" {
      + data_retention_time_in_days = (known after apply)
      + id                          = (known after apply)
      + is_transient                = false
      + name                        = "EUROWIND"
    }
```
To be clear, this is a database which already exists and was deployed with provider 0.59. To replicate this issue, deploy the database with provider 0.59 then upgrade to version 0.60 or 0.61.

**Comment by PedroMartinSteenstrup** [comment]
similar here
```
Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the
last "terraform apply" which may have affected this plan:

  # snowflake_database.this_db["AIRBYTE_AMNIVAC"] has been deleted
  - resource "snowflake_database" "this_db" {
      - data_retention_time_in_days = 1 -> null
      - id                          = "AIRBYTE_AMNIVAC" -> null
      - is_transient                = false -> null
      - name                        = "AIRBYTE_AMNIVAC" -> null
    }

```

```
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # snowflake_database.this_db["AIRBYTE_AMNIVAC"] will be created
  + resource "snowflake_database" "this_db" {
      + data_retention_time_in_days = 1
      + id                          = (known after apply)
      + is_transient                = false
      + name                        = "AIRBYTE_AMNIVAC"
    }
```
But state list contains
```
snowflake_database.this_db["AIRBYTE_AMNIVAC"]
```

**Comment by philippeboyd** [comment]
I have the same issue as @PedroMartinSteenstrup ☝️

**Comment by sfc-gh-swinkler** [comment]
@marcincuber i was not able to replicate this, but i have a fix that i think might work. Can you please take a look and perhaps test it out?

**Comment by mlavaert** [comment]
It's not fixed but the issue is changed. I've created another issue: https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1734

**Comment by marcincuber** [comment]
This issue seems to be resolved with provider version 0.64.0
